plugins/catalog-backend-module-bitbucket-server/src/lib/types.ts

```diff
@@ -33,3 +33,33 @@ export type BitbucketServerRepository = {
 export type BitbucketServerProject = {
   key: string;
 };
+
+/** @public */
+export namespace Events {
+  export interface Event {
+    eventKey: string;
+  }
+
+  export interface PushEvent extends Event {
+    eventKey: string;
+    date: string;
+    actor: Actor;
+    repository: BitbucketServerEventRepository;
+    changes: Change[];
+    commits: undefined;
+    ToCommit: undefined;
+  }
+  export type Actor = {
+    name?: string;
+    id: number;
+  };
+  export type Change = {
+    ref: { id: string; displayId: string; type: string };
+  };
+  export type BitbucketServerEventRepository = {
```

Comment:
Isn't it just a `Repository` and not an `EventRepository`? 🤔 Why don't you use `BitbucketServerRepository` and enhance it with fields that were previously not declared (likely, because they were not needed).

Not all the types seem to be Event-related (e.g., `Actor`).

Reply:
Definitely can change the name, that is much better. I felt it would take a good amount of time to sit down and interface all fields of the event JSON. That's why only some exist, I stopped a little bit into it because I realized how much time it would take. I mainly tried to include only the pieces I would need. Might be better to enhance the rest of the fields as a later follow-up since this specific plugin won't need as of right now? What are your thoughts?